tests/acceptance/01_vars/02_functions/variablesmatching.cf

```diff
@@ -21,25 +25,15 @@ bundle agent test
       "x_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "x");
       "z_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "z");
 
-      "count" int => length(vars);
-      "x_count" int => length(x_vars);
-      "z_count" int => length(z_vars);
+      "fullvars" data => data_matching("default:init.test_b37ce1b03955522848f0a82d0b2b2e21.*");
```

Comment:
(1) It matches the argument with the variables' content, but because more than one variable can be found, the result is a key-value map with the variable names as keys.

(2) This is useful because everything is converted to data containers. The other way you are thinking of doesn't just double-dereference variable names, it also needs know if they are strings or lists or maps, so it's very hard to extract the data.